src/snowflake/connector/reauth_by_plugin.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2012-2021 Snowflake Computing Inc. All rights reserved.
+#
+from abc import ABC, abstractmethod
+from typing import Dict, Optional
+
+
+class ReauthByPlugin(ABC):
+    def __init__(self, conn: Optional["SnowflakeConnection"] = None) -> None:
```

Comment:
Please add `from __future__ import annotations` and use the new type-hint standard
```suggestion
    def __init__(self, conn: SnowflakeConnection | None = None) -> None:
```
I'll add this to our pre-commit config to be added automatically to all files once we merge your PR

Reply:
On a second thought leave it and I'll update all files once I added it to pre-commit